presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -113,6 +114,31 @@ public static Slice stAsText(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return Slices.utf8Slice(deserialize(input).asText());
     }
 
+    @SqlNullable
+    @Description("Returns the geometry that represents all points whose distance from the specified geometry is less than or equal to the specified distance")
+    @ScalarFunction("ST_Buffer")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice stBuffer(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(DOUBLE) double distance)
+    {
+        if (distance < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "distance must be greater than or equal to 0");
+        }
+
+        if (distance == 0) {
+            return input;
+        }
+
+        if (Double.isNaN(distance)) {
```

Comment:
let's move this `isNan` check above the `distance == 0` check so that all failure case checks are together and at the beginning.